lib/galaxy/tools/parameters/output_collect.py

```diff
@@ -23,9 +24,99 @@
 log = logging.getLogger( __name__ )
 
 
+class NullToolProvidedMetadata(object):
+
+    def get_new_datasets(self, output_name):
+        return []
+
+    def get_new_dataset_meta_by_basename(self, output_name, basename):
+        return {}
+
+
+class LegacyToolProvidedMetadata(object):
+
+    def __init__(self, job_wrapper, meta_file):
+        self.job_wrapper = job_wrapper
+        self.tool_provided_job_metadata = []
+
+        with open( meta_file, 'r' ) as f:
+            for line in f:
+                try:
+                    line = json.loads( line )
+                    assert 'type' in line
+                except:
```

Comment:
Yeah - this was copied and pasted from existing code but definitely I should fix that. I will do that.